haystack/preview/components/preprocessors/text_document_splitter.py

```diff
@@ -0,0 +1,109 @@
+from typing import List, Optional, Dict, Any, Literal, Tuple
+
+from more_itertools import windowed
+
+from haystack.preview import component, Document, default_from_dict, default_to_dict
+
+
+@component
+class TextDocumentSplitter:
+    """
+    Split a text document into a list of text documents with shorter texts.
+    This is useful for splitting documents with long texts that otherwise would not fit into the maximum text length of language models
+    """
+
+    def __init__(
+        self, split_by: Literal["word", "sentence", "passage"] = "word", split_length: int = 200, split_overlap: int = 0
+    ):
+        """
+        :param split_by: The unit by which the document should be split. Choose from "word", "sentence", "passage".
+        :param split_length: The maximum length of each split.
+        :param split_overlap: The number of units that each split should overlap.
+        """
+        self.split_by = split_by
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        document: Document,
```

Comment:
I am really trying not to contradict myself too much on this topic, but: let's take a list of Documents here :see_no_evil: 

The fact is that I don't see almost any scenario where this component can be used on a single document. Normally this component is expected to process a list of documents coming from one or more file converters, which all output lists (see any of the components here: https://github.com/deepset-ai/haystack/tree/main/haystack/preview/components/file_converters).

It's also possible that the converters are all wrong! They've been made one from the blueprint of the other. However, in order to connect this with a converter we need to either change this input to a list or we need to change the converters ouput. In turn, changing the converters requires us to change the LinkContentFetcher, which in turn means that it's hard to connect it with the WebSearch component...

Note that making this component take a list as input does *not* affect the output type. That stays a flat list.

Happy to discuss this offline too. There might be a better solution to this problem.

Reply:
Alright, List[Document] is now the input and output type. Still having the concept of indexing pipelines in mind, that makes more sense then single Document as input and we can introduce support for single Document etc. later with a couple of different options.